frontend/src/lib/notification.tsx

```diff
@@ -0,0 +1,34 @@
+import getUuidByString from 'uuid-by-string';
+
+export enum NOTIFICATION_TYPE {
+  DIALOG = 'DIALOG',
+  SNACKBAR = 'SNACKBAR',
+  BOTH = 'BOTH',
+}
+
+export class Notification {
+  type: NOTIFICATION_TYPE = NOTIFICATION_TYPE.DIALOG;
+  message: string = '';
+  id: string;
+  seen: boolean = false;
+  url?: string;
+  date: string = new Date().toUTCString();
+
+  constructor(type?: NOTIFICATION_TYPE, message?: string, date?: string) {
+    if (type) {
+      this.type = type;
+    }
+    if (message) {
+      this.message = message;
+    }
+    if (date) {
+      this.date = date;
+    }
+    // generate the id based on the message and the date attached to a notification
+    this.id = getUuidByString(`${date},${message}`);
+  }
+
+  generateID() {
```

Comment:
This isn't used. Additionally, there's no need for it to be a UUID.

Reply:
This would be a helper within the class that will allow plugin developer to create ids on notification when they use this class

Reply:
I was thinking of adding this earlier but i don't think there will ever be need from outside to change the ID of a notification

Reply:
Seems this was removed. Resolving this conversation.